library/src/main/java/com/novoda/downloadmanager/CompletedDownloadBatchesExtractor.java

```diff
@@ -90,39 +87,36 @@
                             fileIds.add(originalFileId);
                         }
 
-                        if (originalFileId == null) {
-                            newBatchBuilder.downloadFrom(originalNetworkAddress)
-                                    .apply();
-                        } else {
-                            newBatchBuilder.downloadFrom(originalNetworkAddress)
```

Comment:
Confusing to use the `BatchBuilder` which is used for queueing downloads, instead we have a dedicated object.